gcrozariol/gympass-api
# Create all GymPass App features + tests

- Create docker-compose file + add users migration
- Create check ins and gyms tables + relationships
- Apply repository pattern + dependency inversion for register feature
- Add register use case unit tests
- Setup coverage tests + vitest ui
- Add authentication use cases + controller + unit tests
- Create factories for both register and authenticate use cases
- Create get user profile use case
- Add check-in use case
- Add check in use case + in-memory gyms repository
- Add create gym use case
- Add user's check-ins activity
- Add total amount of user's check-ins within metrics use case
- Add search gyms use case
- Add fetch nearby gym use case
- Add check in validation use case
- Minor refactoring - prisma users repository
- Add prisma check-ins repository
- Add prisma gyms repository
- Make check-in use case factory
- Make create gym use case factory
- Make fetch nearby gyms use case factory
- Make fetch user check-ins history use case factory
- Make get user metrics use case factory
- Make get user profile use case factory
- Make search gyms use case factory
- Make validate check in use case factory
- Minor refactoring
- Register fastify jwt
- Add fastify jwt type declaration
- Add middleware to vefiry jwt
- Add prisma test environment
- Finish setup prisma test environment
- Add scripts for e2e tests + e2e pre-tests
- Setup prisma test environment
- Add register e2e tests
- Add authentication e2e tests
- Add profile e2e tests
- Refactor routes
- Add create gym controller
- Add seach gyms controller
- Add nearby gyms controller + setup gyms routes
- Add create check-ins controller
- Add history check-ins controller
- Add metrics check-ins controller
- Add validate check-ins controller
- Setup check-ins routes
- Add test utility to create and authenticate user
- Create gym e2e test
- Search gyms e2e test
- Fetch nearby gyms e2e test
- Create check-ins e2e test
- Get history check-ins e2e test
- Metrics check-ins e2e test
- Validate check-ins e2e test
- Setup refresh token
- Setup refresh token controller + tests
- Add RBAC + implement middleware to create gym and validate check in
- Refactor tests to validate endpoints only admins can access
- Create unit tests workflow
- Update README file
- Create E2E tests workflow
